src/uct/api/uct.h

```diff
@@ -1009,6 +1009,68 @@ ucs_status_t uct_wakeup_signal(uct_wakeup_h wakeup);
 
 /**
  * @ingroup UCT_RESOURCE
+ * @brief Disable synchronous progress for the interface
+ *
+ * Notify the transport that it should avoid doing anything
+ * during @ref uct_worker_progress(). Thus the latency of
+ * other transports may be reduced.
+ *
+ * To progress the interface the application must call
+ * @ref uct_iface_async_progress(). 
+ * 
+ * The function can be called from any context or thread.
+ *
+ * By default the progress is enabled when interface is created.
+ *
+ * @param [in]  iface    The interface to disable progress.
+ *
+ */
+void uct_iface_progress_disable(uct_iface_h iface);
+
+
+/**
+ * @ingroup UCT_RESOURCE
+ * @brief Enable synchronous progress for the interface
+ *
+ * Notify the transport that it should do work 
+ * during uct_worker_progree(). 
+ * Thus the latency of the transport may be reduced.
+ *
+ * The function can be called from any context or thread.
+ *
+ * By default the progress is enabled when interface is created.
+ *
+ * @param [in]  iface    The interface to enable progress.
+ *
+ */
+void uct_iface_progress_enable(uct_iface_h iface);
+
+
+/**
+ * @ingroup UCT_RESOURCE
+ * @brief Progress the interface from any context or thread
+ *
+ * This routine explicitly progresses asynchrounous 
+ * @ref UCT_AM_CB_FLAG_ASYNC active message requests on 
+ * the interface. It may also schedule a slow path callback
+ * @ref uct_worker_slowpath_progress_register to progress
+ * pending requests and sync @ref UCT_AM_CB_FLAG_SYNC
+ * active message requests. 
```

Comment:
I suggest the following re-wording:
 * This routine explicitly progresses asynchronous active message
 * requests, indicated by the @ref UCT_AM_CB_FLAG_ASYNC flag, on
 * the given interface, @a iface. It may also schedule a slow path
 * callback via @ref uct_worker_slowpath_progress_register to progress
 * pending requests and synchronous active message requests,
 * indicated by the @ref UCT_AM_CB_FLAG_SYNC flag.